presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -1756,6 +1757,83 @@ private void testPartitionPerScanLimit(Session session, HiveStorageFormat storag
         assertFalse(getQueryRunner().tableExists(session, tableName));
     }
 
+    @Test
+    public void testPartitionPerScanLimitSessionProperty()
+    {
+        TestingHiveStorageFormat storageFormat = new TestingHiveStorageFormat(getSession(), HiveStorageFormat.DWRF);
+        testWithStorageFormat(storageFormat, this::testPartitionPerScanLimitSessionProperty);
+    }
+
+    private void testPartitionPerScanLimitSessionProperty(Session session, HiveStorageFormat storageFormat)
```

Comment:
nit: looks like this method is used only once. May be fold this into the test method?

Reply:
As far as I understand, testWithStorageFormat is a wrapper and makes it easy to test multiple Hive underlying format, such as CSV. In this case, I think we should leverage it to test all the different Hive storage format underneath. Please correct me if I'm wrong.

Reply:
I meant this ```testWithAllStorageFormats((session, storageFormat) -> {
              // test code
        });```.